lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,6 +37,21 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if (not isinstance(text, str) or
+            not (isinstance(start_of_word, str) or start_of_word is None) or
+            not isinstance(end_of_word, str)):
+        return None
+
+    dict_of_freq = {}
+    words = text.split()
+    for word in words:
+        prepr_word = prepare_word(word, None, '</s>')
```

Comment:
are you sure? Why do not you use arguments?

Reply:
fixed